src/bag.c

```diff
@@ -0,0 +1,490 @@
+{
+    for (u32 i = 0; i < size - 1; i++) {
+        for (u32 j = i + 1; j < size; j++) {
+            if (pocket[i].quantity == 0) {
+                BagItem_Swap(&pocket[i], &pocket[j]);
+            }
+        }
+    }
+}
+
+void Pocket_Sort(BagItem *pocket, const u32 size)
+{
+    for (u32 i = 0; i < size - 1; i++) {
+        for (u32 j = i + 1; j < size; j++) {
+            if (pocket[i].quantity == 0 || (pocket[j].quantity != 0 && pocket[i].item > pocket[j].item)) {
+                BagItem_Swap(&pocket[i], &pocket[j]);
+            }
+        }
+    }
+}
+
+void *sub_0207D824(Bag *bag, const u8 *pockets, enum HeapId heapID)
+{
+    UnkStruct_0207CB08 * v0;
+    int i;
+
+    v0 = sub_0207CB08(heapID);
+
+    for (i = 0; pockets[i] != 0xff; i++) {
+        switch (pockets[i]) {
+        case POCKET_KEY_ITEMS:
+            sub_0207CB48(v0, bag->keyItems, POCKET_KEY_ITEMS, i);
+            break;
+        case POCKET_ITEMS:
+            sub_0207CB48(v0, bag->items, POCKET_ITEMS, i);
+            break;
+        case POCKET_BERRIES:
+            sub_0207CB48(v0, bag->berries, POCKET_BERRIES, i);
+            break;
+        case POCKET_MEDICINE:
+            sub_0207CB48(v0, bag->medicine, POCKET_MEDICINE, i);
+            break;
+        case POCKET_BALLS:
+            sub_0207CB48(v0, bag->pokeballs, POCKET_BALLS, i);
+            break;
+        case POCKET_BATTLE_ITEMS:
+            sub_0207CB48(v0, bag->battleItems, POCKET_BATTLE_ITEMS, i);
+            break;
+        case POCKET_MAIL:
+            sub_0207CB48(v0, bag->mail, POCKET_MAIL, i);
+            break;
+        case POCKET_TMHMS:
+            sub_0207CB48(v0, bag->tmHms, POCKET_TMHMS, i);
+            break;
+        }
+    }
+
+    return v0;
+}
+
+BagItem *Bag_GetItemSlot(Bag *bag, u16 pocketID, u16 slot)
+{
+    BagItem *pocket;
+    u16 pocketSize;
+
+    switch (pocketID) {
+    case POCKET_KEY_ITEMS:
+        pocket = bag->keyItems;
+        pocketSize = KEY_ITEM_POCKET_SIZE;
+        break;
+    case POCKET_ITEMS:
+        pocket = bag->items;
+        pocketSize = ITEM_POCKET_SIZE;
+        break;
+    case POCKET_BERRIES:
+        pocket = bag->berries;
+        pocketSize = BERRY_POCKET_SIZE;
+        break;
+    case POCKET_MEDICINE:
+        pocket = bag->medicine;
+        pocketSize = MEDICINE_POCKET_SIZE;
+        break;
+    case POCKET_BALLS:
+        pocket = bag->pokeballs;
+        pocketSize = POKEBALL_POCKET_SIZE;
+        break;
+    case POCKET_BATTLE_ITEMS:
+        pocket = bag->battleItems;
+        pocketSize = BATTLE_ITEM_POCKET_SIZE;
+        break;
+    case POCKET_MAIL:
+        pocket = bag->mail;
+        pocketSize = MAIL_POCKET_SIZE;
+        break;
+    case POCKET_TMHMS:
+        pocket = bag->tmHms;
+        pocketSize = TMHM_POCKET_SIZE;
+        break;
+    }
+
+    return slot >= pocketSize ? NULL : &pocket[slot];
+}
+
+Bag *SaveData_GetBag(SaveData *saveData)
+{
+    return SaveData_SaveTable(saveData, 3);
```

Comment:
**polish:** there should be an enum value in `constants/savedata/save_table.h` that you can use here